packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol

```diff
@@ -260,13 +299,15 @@ abstract contract CrossDomainMessenger is
         uint256 _value,
         uint256 _minGasLimit,
         bytes calldata _message
-    ) external payable nonReentrant whenNotPaused {
+    )
+        external
+        payable
+        perMessageNonReentrant(
+            Hashing.hashCrossDomainMessage(_nonce, _sender, _target, _value, _minGasLimit, _message)
+        )
+        whenNotPaused
+    {
         (, uint16 version) = Encoding.decodeVersionedNonce(_nonce);
-        require(
```

Comment:
Why is assertion removed? It should be impossible to prove a >2 message to be honest so it might be safe to remove

Reply:
The call to `hashCrossDomainMessage` in the `perMessageNonReentrant` will revert if the version is not `0` or `1`, so it's no longer necessary.

Reply:
I don't love that this is implicit now. Same comment as above that it'd be much simpler if this were just a mapping like `mapping (bytes32 => bool) public reentrancy`.